solver/llbsolver/history.go

```diff
@@ -680,6 +680,8 @@ func (h *HistoryQueue) Update(ctx context.Context, e *controlapi.BuildHistoryEve
 	h.mu.Lock()
 	defer h.mu.Unlock()
 
+	e = e.CloneVT()
```

Comment:
Should this happen outside of the lock?

Reply:
The race is not with this function itself. It is that the record gets saved in the `actives` and remains there after this function has returned.

Reply:
It doesn't which is why I approved it but it might benefit to perform an action that doesn't require the lock outside of the lock so it doesn't hold the lock as long.

Probably not a hotspot though so doesn't matter much.